scripts/drs_file_access_examination.ipynb

```diff
@@ -0,0 +1,326 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Resolve DRS URIs and Checking File Access\n",
+    "\n",
+    "DRS creates a unique ID mapping that allows for flexible file retrieval. The unique mapping is the DRS Uniform Resource Identifier (URI) - a string of characters that uniquely identifies a particular cloud-based resource (similar to URLs) and is agnostic to the cloud infrastructure where it physically exists. To learn where the file physically exists on the cloud, we must resolve the DRS through a backend service called Martha that will unmap the DRS url to get the file's google bucket file path.\n",
+    "\n",
+    "In this step, we check if all the DRS urls can be resolved and check our access to the file it points to in the google bucket. If the DRS url can't be resolved, we will record the error that will, later on, be shown in the report in the last step.\n",
+    "\n",
+    "One of the most comment reasons the DRS urls can not resolve is \"Fence is not linked\". If you have this error, it probably because the data is controlled-access. To use controlled-access data on Terra, you will need to link your Terra user ID to your authorization account (such as a dbGaP account). Linking to external servers will allow Terra to automatically determine if you can access controlled datasets hosted in Terra (ex. TCGA, TOPMed, etc.) based on your valid dbGaP applications. Go to this link to learn more more: https://support.terra.bio/hc/en-us/articles/360038086332\n",
+    "\n",
+    "## Estimated time \n",
+    "\n",
+    "This code resolved and check file access to 50 DRS url a 1 min\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import requests\n",
+    "import sys\n",
+    "from subprocess import Popen, PIPE\n",
+    "from terra_notebook_utils import drs\n",
+    "import json\n",
+    "\n",
+    "# Getting the access_token\n",
+    "access_token = !gcloud auth print-access-token\n",
+    "\n",
+    "# Assigning the count variable to keep track of resolved DRS urls\n",
+    "resolved_DRS_urls_count = 0\n",
+    "\n",
+    "# Assigning the unresolved DRS urls errors list variable\n",
+    "unresolved_DRS_urls_errors_list = []\n",
+    "\n",
+    "# Assigning the files without access errors list variable\n",
+    "files_without_access_errors_list = []\n",
+    "\n",
+    "# Outputting the start of resolving DRS urls\n",
+    "print(\"Resolving \"+ str(len(DRS_urls_dict)) + \" DRS urls\")\n",
+    "\n",
+    "# Assigning the count variable to keep track files with access\n",
+    "files_with_access = 0\n",
+    "\n",
+    "# Assigning the count variable to keep track files without access\n",
+    "files_without_access = 0\n",
+    "\n",
+    "\n",
+    "# Going through each DRS url in the master DRS urls dictionary \n",
+    "for DRS_url in DRS_urls_dict.keys():\n",
```

Comment:
if you use this `for DRS_url, DRS_url_information in DRS_urls_dict.items():` then you can delete the next 2 lines